web_src/js/features/repo-issue-pr-status.js

```diff
@@ -2,11 +2,21 @@ export function initRepoPullRequestCommitStatus() {
   for (const btn of document.querySelectorAll('.commit-status-hide-checks')) {
     const panel = btn.closest('.commit-status-panel');
     const list = panel.querySelector('.commit-status-list');
+    const header = panel.querySelector('.commit-status-header');
+    let hasScrollBar = false;
```

Comment:
This change is not right. The annoying scrollbar appears again during the transition when the list is short.

If it's difficult to understand and you can bear the UI flickering caused by the scrollbar, just remove this fine-tune.

Reply:
I have overlay scrollbars on mac so it doesn't bother me that the scrollbar is visible during transition. Maybe show a video of what you mean?

Reply:
I guessed that you don't like flicker, so I added these tweaks. If it doesn't bother you, I will remove them.

Reply:
Looks ok to me for the scrollbar to briefly appear:

![image](https://github.com/go-gitea/gitea/assets/115237/60607013-0936-4166-b8bc-b128c9affcdf)